### DISCUSSION TITLE
Filtering input files for batch execution?

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#23] [Ideas]
In Sweden there is now an option where you download lidar files to get also a border of lidar-files outside an area. The nice thing about that is that you can get vegetation all teh way to the border and not get white stripes in yellow of green.

The easy way to do this is to just execute batch on all files, teh ones you really want to process and the border lidar-files. To being able to sort the output file I normally rename the border-laz-files so they get a prefix "borders_". Then I can just throw away the output files that contains "borders_". 

Simple solution, but it consumes execution time.

Would it be possible to filter the input files so the laz-list does not contain files with a certain prefix?

I mean, I think it is easy to do so by altering the line 1640 in pullauta perl from 

system( 'ls -1 ' . $lazfolder . '*.laz > lazlist' . $thread . '.txt' );
 
to

system('ls -1 ' . $lazfolder . '*.laz | grep -v "^' . $lazfolder . 'borders_" > lazlist' . $thread . '.txt');

But what I do not know is if that also will affect how the yellow/green is rendered at the border? I.e. are the border laz-files considered when the yellow green is calculated.

**Comment by rphlo** [comment]
So you would like files with a given prefix to only go to makevegetation step and skip contours and other steps?

**Comment by stefankinell** [comment]
Honestly since I do not know how the program is built up I'm not sure.

I was assuming that when making vegetation kartapullautin checks if there are neighbouring laz-files and if there is - it uses the information to extend the vegetation all to the border. This is what happens for me when I run a set of laz-files with border laz-files in the In folder. 

So when I run the larger areas in batch mode I thought by adding the border laz-files to the in-folder I would create vegetation files that extend the whole way to the sides, but in the end I then just throw away the output files from the border laz-files since they have no use for me. So my thought was if it was possible to skip the processing of them but just use them in the makevegetation step.

I'll see if I can understand the code and if my guess is right on how adjacent laz-files are used in the vegetation step.

On another note - being able to only run makevegetation on some (or all) files in batch-mode has been a wish we talked about earlier, but I agreed on let's make the basics work of the original KP before adding new functions.

**Comment by stefankinell** [comment]
I fixed this outside the pullouta-script. 

In Sweden, when you download a full set of a lax-area, typically 200 laz-files, you can also request to get "border" laz-files. Which is as  the name suggests a set of laz-files that border the real area.

So my method is now to:
1. rename all border-laz-files so they have a prefix borders_*.laz
2. look at my input folder for all files named borders_* and create an empty .png-file in the output folder.

This way the laz-files are used in the pullouta-script so that vegetation is rendered all the way to the border, but time is not spent on actually creating output files for the border sections.

If anybody needs this I can share the python-sctipt I built to sort the input files, and create the .png-files for the borders_ files. 

With this I do not see any reason for creating more complexity in the pullouta-sctipt when it can be solved in a simple way in preparation.
